wfc_gh/WaveFunctionCollapse.cs

```diff
@@ -444,22 +464,42 @@ protected override void SolveInstance(IGH_DataAccess DA)
             bool foundDeterministic = false;
             uint attempts = 0;
 
-            while (!foundDeterministic && attempts < maxAttempts)
-            {
-                var result = Native.wfc_observe(wfcWorldStateHandle, wfcRngStateHandle);
+            uint limitObservations = 0;
+            uint maxObservations = UInt32.MaxValue;
+
+            var spentObservations = IntPtr.Zero;
```

Comment:
I am fairly sure this will crash. Did you test it?

Reply:
It works as expected... Why would it crash?

Reply:
I am testing it [here](https://github.com/subdgtl/Monoceros/pull/26) though.

Also, is it still feasible to maintain the Grasshopper part in this repo?

Reply:
> It works as expected... Why would it crash?

Rust asserts the pointer is not null and you pass something that looks like a null pointer in C#. The use of `IntPre.Zero` confused me, but I'll address that in another review comment.

> Also, is it still feasible to maintain the Grasshopper part in this repo?

3 out of 4 times I needed to do something in the Monoceros repo, I had trouble building it (for various reasons, e.g. missing common DLLs, git LFS, etc.).  I'd consider droppping `wfc_gh` once the development experience in Monoceros repo is seamless. The reason for keeping it is that it is much simpler than Monoceros and it might serve as documentation or inspiration for someone to either build their own GH plugin.

Reply:
I've got stories with trying to make Rust work too... Not sure it should be a benchmark.

Reply:
Also this works both ways - long story short: shouldn't I include the compiled `dll` from here to the Monoceros repo to make it work without the need to deploy this repo (which is not a simple task)?

Reply:
Yes, the solver DLL should be vendored in Monoreros repo. Some care needs to be taken so that it is always the correct one, though.

Reply:
Btw what difficulties did you have? Did something not work according to the readme?